Functions/xASL_bids_CreateDatasetDescriptionTemplate.m

```diff
@@ -0,0 +1,105 @@
+function [json] = xASL_bids_CreateDatasetDescriptionTemplate(varargin)
+%xASL_bids_CreateDatasetDescriptionTemplate This script creates a JSON structure which can be saved
+% using spm_jsonwrite to get a dataset_description.json template.
+%
+% FORMAT: [json] = xASL_bids_CreateDatasetDescriptionTemplate(varargin)
+% 
+% INPUT:
+%   Name                - Patient name (CHAR ARRAY, OPTIONAL, DEFAULT = 'Undefined')
+%   BIDSVersion         - BIDS Version (e.g. 1.5.0) (CHAR ARRAY, OPTIONAL, DEFAULT = use xASL_bids_Config to get version)
+%   HEDVersion          - HED Version (CHAR ARRAY, OPTIONAL, DEFAULT = 'Undefined')
+%   DatasetType         - Dataset Type (CHAR ARRAY, OPTIONAL, DEFAULT = 'Undefined')
+%   License             - License (CHAR ARRAY, OPTIONAL, DEFAULT = 'Undefined')
+%   Authors             - Authors (CHAR ARRAY, OPTIONAL, DEFAULT = 'Undefined')
+%   Acknowledgements    - Acknowlegements (CHAR ARRAY, OPTIONAL, DEFAULT = 'Undefined')
+%
+% OUTPUT:
+%   json                - JSON structure which can be saved using spm_jsonwrite (dataset_description.json template)
+%                         
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:  This script creates a JSON structure which can be saved
+%               using spm_jsonwrite to get a dataset_description.json template.
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:      [json] = xASL_bids_CreateDatasetDescriptionTemplate('Test Patient','1.5.0','1.2.3','Test','Test License','Author Names','Thanks');
+%               [json] = xASL_bids_CreateDatasetDescriptionTemplate();
+%               
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    % Get default BIDS configuration
+    bidsPar = xASL_bids_Config();
+
+    % Create dummy dataset_description.json
+    json = struct;
+    
+    % Required fields
+    for iCell = 1:numel(bidsPar.datasetDescription.Required)
+         json.(bidsPar.datasetDescription.Required{1,iCell}) = 'Undefined';
+    end
+    % Recommended fields
+    for iCell = 1:numel(bidsPar.datasetDescription.Recommended)
```

Comment:
Recommended and optional fields should be rather not assigned at all - i.e. skip them in the struct instead of assigning 'undefined' or ''.

Reply:
But overall, it is great that you use the 'required' and 'recommended' lists from the BIDS config - it will make our live easier during an update :)

Reply:
Check out bcfe3e6.